test/unit/spec/PasswordExpired_spec.js

```diff
@@ -124,14 +128,57 @@ function (Q, _, $, OktaAuth, LoginUtil, Util, PasswordExpiredForm, Beacon, Expec
             username: 'inca@clouditude.net',
             password: 'newpwd'
           });
+          return tick();
+        }).then(function() {
           expect($.ajax.calls.count()).toBe(1);
         });
       });
+      itp('calls async processCreds function before saving a model', function () {
+        return setup({
+          'processCreds': function(creds, callback) {
+            processCredsAsyncSpy(creds, callback);
+            callback();
+          }
+        }).then(function (test) {
+          $.ajax.calls.reset();
+          processCredsAsyncSpy.calls.reset();
+          test.setNextResponse(resSuccess);
+          submitNewPass(test, 'oldpwd', 'newpwd', 'newpwd');
+          expect(processCredsAsyncSpy.calls.count()).toBe(1);
+          expect(processCredsAsyncSpy).toHaveBeenCalledWith({
+            username: 'inca@clouditude.net',
+            password: 'newpwd'
+          }, jasmine.any(Function));
+          return tick();
+        }).then(function() {
+          expect($.ajax.calls.count()).toBe(1);
+        });
+      });
+      itp('calls async processCreds function and can prevent saving a model', function () {
+        return setup({
```

Comment:
Something is wrong here. This file isn't the latest code on the branch:
https://github.com/AlainODea/okta-signin-widget/blob/ao-asyncProcessCreds-Promises/test/unit/spec/PasswordExpired_spec.js#L157